### PR TITLE
Send perf

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] don't thrash inbox on sends
- [x] don't thrash when hiding/showing waypoints
- [x] don't thrash input while typing
- [x] don't cascade error boundary when children change but there's no info change

### CODE REVIEW

File: shared/chat/conversation/messages/wrapper/send-indicator/index.tsx [hunk 1]
```diff
@@ -32,7 +32,24 @@ class SendIndicator extends React.Component<Props, State> {
 
   constructor(props: Props) {
     super(props)
-    this.state = {iconStatus: 'encrypting', visible: !props.sent}
+    const state: State = {iconStatus: 'encrypting', visible: !props.sent}
+
+    if (!(this.props.sent || this.props.failed)) {
```
Comment (MarcoPolo) [inline shared/chat/conversation/messages/wrapper/send-indicator/index.tsx]: `props` instead of `this.props`

Comment (MarcoPolo) [reply shared/chat/conversation/messages/wrapper/send-indicator/index.tsx]: 🐑 I think this is easier to understand if we flip it a la De Morgan's. `!this.props.sent && !this.props.failed`

File: shared/chat/conversation/list-area/normal/index.desktop.tsx [hunk 2]
```diff
@@ -624,11 +624,19 @@ class OrdinalWaypoint extends React.Component<OrdinalWaypointProps, OrdinalWaypo
     return null
   }
 
+  // Cache rendered children if the ordinals are the same, else we'll thrash a lot as we scroll up and down
+  private lastVisibleChildrenOrdinals: Array<Types.Ordinal> = []
+  private lastVisibleChildren: React.ReactNode = null
```
Comment (MarcoPolo) [inline shared/chat/conversation/list-area/normal/index.desktop.tsx]: Shouldn't this be a SCU instead? This seems hacky

File: shared/chat/conversation/messages/wrapper/send-indicator/index.tsx (hunk 1 above)
Comment (chrisnojima) [reply shared/chat/conversation/messages/wrapper/send-indicator/index.tsx]: this is just a movement of the below. also here props === this.props

File: shared/chat/conversation/list-area/normal/index.desktop.tsx (hunk 2 above)
Comment (chrisnojima) [reply shared/chat/conversation/list-area/normal/index.desktop.tsx]: no cause it does change. we just don't need to call render again

**Pull request merged**